### PR TITLE
tooling: Use upstream base libs

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: tooling: Use upstream base libs
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
this obsoletes the README in /tools but i would prefer to deal with that separately - i will shift most of the info to the pytooling repo and update these after, preferably

**Comment by phlax** [comment]
marking this WIP to separate out the changes to api_proto_breaking_change_detector

**Comment by phlax** [comment]
this is still blocked by the problem addressed in #17981

File: tools/base/requirements.in
Comment (KBaichoo) [inline tools/base/requirements.in:7]: Do other maintainers have access to control https://libraries.io/pypi/envoy.base.checker? and https://libraries.io/pypi/envoy.base.runner?

File: tools/base/BUILD
Comment (KBaichoo) [inline tools/base/BUILD:10]: Where exactly is this file? I don't see it in this PR or in the repo as of https://github.com/envoyproxy/envoy/commit/5c4ff96652b449a39cc640a97794333b7bc723d4?

**Review by KBaichoo — COMMENTED** [review]
Thanks for working on this!

File: tools/base/requirements.in
Comment (phlax) [reply tools/base/requirements.in:7]: the packages are maintained here - https://github.com/envoyproxy/pytooling

im mostly adding comments to that effect to any stubs remaining in the Envoy code base so that is clearer

the repo is under project control so, yep, re access

File: tools/base/BUILD
Comment (phlax) [reply tools/base/BUILD:10]: good catch - this should be removed i think

**Review by KBaichoo — APPROVED** [review]
lgtm

**Comment by phlax** [comment]
cc @htuch

**Review by htuch — APPROVED** [review]

**Pull request merged**
